samples/wifi/provisioning/implementation_details.rst

```diff
@@ -0,0 +1,219 @@
+These definitions are available as `.proto` files in the sample code.
+The sample uses `nanopb`_ to convert encoded platform-independent message to C language data type, and vice versa.
+By exchanging messages with configurator (the peer that sends the message, for example, your smartphone that runs nRF Wi-Fi Provisioner app), the device will perform corresponding actions.
+
+message types
+-------------
+
+Four types of message are defined. They are:
+
+1. `Info`. It is to allow the configurator to know the capability of device before actual provisioning process happens.
+   The configurator should read this message before send any other message.
+   Currently, there is only a field of `version`` inside.
+
+#. `Request`. The message contains a command and configuration.
+   The device takes action based on the command and applies the configuration if one is given.
+   Currently, 5 commands are defined.
+
+   The following table details the five defined commands used by the sample.
+
+   =================== ================================================ =============================
+   Command             Operatoin                                        Notes
+   =================== ================================================ =============================
+   GET_STATUS          to get the status of the device, like
+                       provisioning state, Wi-Fi connection state, etc
+   START_SCAN          to trigger AP scan                               scan parameter is optional
+   STOP_SCAN           to stop AP scan
+   SET_CONFIG          to store Wi-Fi configuration and connect to      connection param is required
+                       that AP
+   FORGET_CONFIG       to erase the configuration and disconnect
+   =================== ================================================ =============================
+
+#. `Response`. The message contains the command to which it is responding, the operation result, and any other data.
+   The data associated with this command may be sent to configurator after the `Response` message.
+
+   The `Response` message contains one of following response codes.
+
+   * `SUCCESS` - the operation is dispatched successfully
+   * `INVALID_ARGUMENT` - the argument in Request is invalid
+   * `INVALID_PROTO` - the message cannot be encoded or decoded correctly
+   * `INTERNAL_ERROR` - the task cannot be dispatched properly
+
+#. `Result`. It is used to carry data sent in asynchronous way.
+
+   * If the command is `START_SCAN`, and the scan process is done, the information of AP will be sent to configurator using `Result` message.
+     Each `Result` contains one AP information.
+   * If the command of `SET_CONFIG`, and When the status of Wi-Fi changes (e.g., from disconnected to associated), the new status will be sent to configurator using `Result` message.
+
+You may find the details of definition in .proto files under path of `./common/proto`.
+
+`GET_STATUS` procedure
+----------------------
```

Comment:
```suggestion
* GET_STATUS: retrieves current device status. The response includes some or all of the following fields:
   * `state`: describes the Wi-Fi connection state according to values defined in `common.proto`.
   * `provisioning_info`: includes Wi-Fi provisioning information stored in the device's non-volative memory (NVM).
   * `connection_info`: includes Wi-Fi connection details.
   *  `scan_info`: includes the parameters used for the AP scan.
```

the backticks are rendering the text in italics when it should be monospace. I'm not sure why, but @richabp would have a better idea of formatting.

The idea here is a nested, unordered list.

* Command: description
  * additional parameter: description